src/Infrastructure/GoogleListingsAndAdsPlugin.php

```diff
@@ -40,6 +46,7 @@ final class GoogleListingsAndAdsPlugin implements Plugin {
 	 */
 	public function __construct( ContainerInterface $container ) {
 		$this->container = $container;
+		$this->set_options_object( new Options() );
```

Comment:
Since this is the main class the OptionsAware interface and trait don't really work or make sense. Also we don't generally instantiate new classes if they are provided by the container, we let the container handle whether it should return a new or a shared instance. If options is needed in this class then I'd suggest to create a new private member and then set it from the container.

```suggestion
		$this->options   = $container->get( OptionsInterface::class );
```

Reply:
Ah. Thanks for this. I will correct

Reply:
turns out, I get anything from the container at the constructor level. I decided not to initialize an `options` property.